### PR TITLE
chore: add storage reservation tests

### PR DESCRIPTION
**Pull request by maksymar**
This PR adds tests for storage reservation mechanism.

### CODE REVIEW

File: rs/execution_environment/tests/storage_reservation.rs [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        UNIVERSAL_CANISTER_WASM.to_vec(),
+        vec![],
+    )
+    .unwrap();
+
+    (env, canister_id)
+}
+
+#[test]
+fn test_storage_reservation_not_triggered() {
```
Comment (dsarlis) [inline rs/execution_environment/tests/storage_reservation.rs:47]: Perhaps add a comment on why we expect that the reservation mechanism is not triggered?

Actually, this applies to all tests I guess. Maybe we have some comment that explains kinda the setup. We have a specific threshold and capacity, we expect the mechanism to trigger whenever you try to grow your memory above the threshold but not when you're below.

**Review by dsarlis — COMMENTED** [review]
Should we use `test` commit tag instead of `chore`? We are adding a bunch of tests after all...

File: rs/state_machine_tests/src/lib.rs [hunk 2]
```diff
@@ -3574,6 +3574,21 @@ impl StateMachine {
             .get()
     }
 
+    /// Returns the reserved balance of the specified canister.
+    ///
+    /// # Panics
+    ///
+    /// This function panics if the specified canister does not exist.
+    pub fn reserved_balance(&self, canister_id: CanisterId) -> u128 {
```
Comment (mraszyk) [inline rs/state_machine_tests/src/lib.rs:3582]: Why is a new function needed? Can't you just use the existing canister status endpoint (maybe together with get_controllers if you don't have a controller around)?

File: rs/execution_environment/tests/storage_reservation.rs (hunk 1 above)
Comment (maksymar) [reply rs/execution_environment/tests/storage_reservation.rs:47]: I added high level comments to each test.

File: rs/state_machine_tests/src/lib.rs (hunk 2 above)
Comment (maksymar) [reply rs/state_machine_tests/src/lib.rs:3582]: good point, switched to using canister_status, thank you!

**Review by dsarlis — APPROVED** [review]

**Pull request merged**